neo/Consensus/ConsensusService.cs

```diff
@@ -363,6 +345,28 @@ private void OnRecoveryMessageReceived(ConsensusPayload payload, RecoveryMessage
             }
         }
 
+        private void OnRecoveryRequestReceived(ConsensusPayload payload)
+        {
+            if (context.WatchOnly()) return;
+            if (!context.CommitSent())
+            {
+                bool shouldSendRecovery = false;
+                // Limit recovery to be sent from, at least, `f` nodes when the request is from a lower view number.
+                int allowedRecoveryNodeCount = context.F();
+                for (int i = 1; i <= allowedRecoveryNodeCount; i++)
+                {
+                    var chosenIndex = (payload.ValidatorIndex + i) % context.Validators.Length;
```

Comment:
This is nice Erik.
The only difference I noticed is `var chosenIndex = (payload.ValidatorIndex + i) % context.Validators.Length;` against `var chosenIndex = (payload.ValidatorIndex + i + MESSAGE.VIEWNUMBER) % context.Validators.Length;`
do you think this is relevant for the allowedRecoveryNodeCount logic?

Reply:
I have made a little modification to this logic, you can see if it is reasonable.

Reply:
In fact I misclicked "approve" hahah I'll comment other small things too... just to make sure we do the right thing.